### PR TITLE
Add function to upload data on resource creation

### PR DESCRIPTION
**Pull request by individual-it**
TUS can transfer data already on resource creation (if creation-with-upload﻿ extension is enabled) see https://tus.io/protocols/resumable-upload.html#creation-with-upload﻿
the new function allows us to use that extension in the client

### CODE REVIEW

**Comment by individual-it** [comment]
something is weird with codecov
Line 470 is hit by `it_throws_exception_when_unable_to_create_resource`
Line 461 is hit by `it_creates_a_partial_resource_with_post_request`

File: src/Tus/Client.php
Comment (ankitpokhrel) [inline src/Tus/Client.php:442]: > -1 => all data; 0 => no data

In the implementation below we always check if `$bytes > 0` to decide if we need to send data to the server to upload. In that case, doesn't it make more sense to use zero as a default value for `$bytes`?

**Review by ankitpokhrel — COMMENTED** [review]
@individual-it Thanks a lot for the PR 👍 

Looks like the `creation-with-upload` was [merged](https://github.com/tus/tus-resumable-upload-protocol/pull/88#event-2690108871) to the protocol not so long ago.

I added some minor comments, it looks good otherwise.

The code coverage issue is because of the missing `@covers` annotation in some tests. From the PHPUnit doc:

> If provided, this effectively filters the code coverage report to include executed code from the referenced code parts only. This will make sure that code is only marked as covered if there are dedicated tests for it, but not if it used indirectly by the tests for a different class, thus avoiding false positives for code coverage.

Adding `@covers ::createWithUpload` in the doc-block of the following tests should fix the issue with codecov.
```
it_throws_exception_when_unable_to_create_resource
it_creates_a_partial_resource_with_post_request
it_throws_exception_when_unable_to_create_resource
```

File: src/Tus/Client.php
Comment (individual-it) [reply src/Tus/Client.php:442]: It's to be consistent with the `upload` function.
And to use that function with `$bytes = 0` does not make any sense, then you would use the `create` function. The option to send `$bytes = 0` is really only there to reuse the code also for the `upload` function.

It's your call, whatever you prefer more.

File: src/Tus/Client.php
Comment (individual-it) [reply src/Tus/Client.php]: good point, done in 8b8d165b3520998ec3c6305cd40785e7da70dac5

File: src/Tus/Client.php
Comment (individual-it) [reply src/Tus/Client.php:455]: true, saves one if statement. done in 5d1ffbd2464145fefe75d1450dd96e3ca49b6ede

File: tests/Tus/ClientTest.php
Comment (individual-it) [reply tests/Tus/ClientTest.php]: made it look nicer in 3050fcf1c94de41abb27339511df4972348aea50

**Comment by individual-it** [comment]
@ankitpokhrel thank you for the review. I've changed the most things and also to happy to adjust the code according to your last comment if you want it to have it that way.
To make it easier to review again I've created single commits, but very happy to squash them before merging, whatever your policy in this repo is

File: src/Tus/Client.php
Comment (ankitpokhrel) [reply src/Tus/Client.php:442]: I got your point now, negative default value is fine. Thanks!

**Review by ankitpokhrel — APPROVED** [review]

**Comment by swoichha** [comment]
@ankitpokhrel can you make a new release, so that I can use `createWithUpload` function?

**Comment by ankitpokhrel** [comment]
Sorry, the new release is available now: v1.3.1 and v2.1.2

**Pull request merged**
